### ISSUE TITLE
Update to rc5

### ISSUE THREAD

**Issue by gnujeremie** [ocombe/ng2-translate#187]
The rc5 introduces the concept of module (@NgModule).

**Comment by ocombe** [comment]
yep it's on my todo list, in the mean time you can still make ng2-translate work by using `providers` in your ng module to provide the lib :)

**Comment by gnujeremie** [comment]
thanks, that's what i was looking for ^^

**Comment by moh3n9595** [comment]
i want to use translator in rc5 but i can't. please explain your  solution step by step ,earlear.because  i have time limitation for Delivery of the project please

**Comment by ocombe** [comment]
It's simple, previously in RC4 you would define the lib at bootstrap like this:

```ts
bootstrap(AppComponent, [
    HTTP_PROVIDERS,
    // not required, but recommended to have 1 unique instance of your service
    TRANSLATE_PROVIDERS
]);
```

Now you must define it in your @NgModule like this:
```ts
@NgModule({
    declarations: [
        // global pipes
        TranslatePipe // optional, just if you want it to be available everywhere
    ],
    providers: [
        TRANSLATE_PROVIDERS
    ],
    imports: [
        BrowserModule,
        HttpModule
    ],
    bootstrap: [App],
})
export class AppModule {
}
```

**Comment by moh3n9595** [comment]
tnx a lot dear, I tested it! but it does not  work,  i don't get any error in browser and just see "loading.." :(

**Comment by gnujeremie** [comment]
translation is working for me in rc5. are you sure there are no logs in the browser console ?

**Comment by ocombe** [comment]
Did you change your bootstrap to load your new NgModule like explained here: https://angular.io/docs/ts/latest/cookbook/rc4-to-rc5.html ?

**Comment by moh3n9595** [comment]
yes i checked both. 
my appModule : 
`
//some imports..
@NgModule({
  imports: [
    BrowserModule,
    FormsModule,
    routing,
    HttpModule,
  ],
  declarations: [
    AppComponent,
    DashboardComponent,
     TranslatePipe
  ],
  providers: [
     TRANSLATE_PROVIDERS
  ],
  bootstrap: [ AppComponent ]
})

export class AppModule {
}`

my main : 
`import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
import { AppModule }              from './app.module';

platformBrowserDynamic().bootstrapModule(AppModule);`

my appcomponent: 
`export class AppComponent {

constructor(private translate: TranslateService ) {
 var userLang = navigator.language.split('-')[0]; // use navigator lang if available
        userLang = /(fa|en)/gi.test(userLang) ? userLang : 'fa';
        translate.setDefaultLang('fa');      
        translate.use(userLang);
      translate.use('fa');

}`
do you have any solution?

**Comment by ocombe** [comment]
It looks fine to me, I don't think the problem comes from ng2 translate, if it was the cause you would get an error about "No provider" or "No FactoryMetadata"

**Comment by moh3n9595** [comment]
tnx dear, i check again and if i find any solution i will say it here! :)

**Comment by moh3n9595** [comment]
@ocombe hi again! your code is right. but i have a question about formbuilder from you because i think you know answer! and my question is : what does work formbuilder in rc5 ? (i know it in rc4)

**Comment by ocombe** [comment]
Sorry I haven't had the occasion to do forms in a while, I don't think I can help you on that :)

**Comment by spock123** [comment]
Hey guys,
- could it be because ng2translate uses the http service? 

I'm getting the following error when using the StaticFileLoader: 

"zone.js:463 Error: Uncaught (in promise): TypeError: this.http.get is not a function(…)"

**Comment by spock123** [comment]
@moh3n9595  I have FormsModule working in RC5.

You need to import either the FormsModule or ReactiveFormsModule (or both).
If you use FormBuilder, then I believe you'll have to use ReactiveFormsModule. 

Try it.. but it really doesn't belong in the ng2-translate repo :)

**Comment by jusefb** [comment]
Can't get this to work. I am trying to use the TranslatePipe in two modules. So I have declared a shared module called TranslateModule:
```
import {NgModule} from "@angular/core";
import {
    TranslatePipe, MissingTranslationHandler, TranslateLoader, TranslateStaticLoader,
    TranslateService
} from "ng2-translate";
import {SpMissingTranslationHandler} from "./utils/missingtranslation.handler";
import {Http, HttpModule} from "@angular/http";
import {BrowserModule} from "@angular/platform-browser";

@NgModule({
    declarations:[
        TranslatePipe
    ],
    providers:[
        {
            provide: MissingTranslationHandler,
            useClass: SpMissingTranslationHandler
        },
        {
            provide: TranslateLoader,
            useFactory: (http: Http) => new TranslateStaticLoader(http, 'dist/assets/i18n', '.json'),
            deps: [Http]
        },
        TranslateService
    ],
    imports:[
        BrowserModule,
        HttpModule
    ]
})
export class TranslateModule{}
```

This module is then imported in the other two modules. For some reason I get a template parse error saying that the TranslatePipe is not available "Error: Template parse errors:↵The pipe 'translate' could not be found ".

Hope someone will be able to help

**Comment by spock123** [comment]
You need to import CommonModule, not Browsermodule when creating shared
modules.

**Comment by spock123** [comment]
Also, remember to put components and pipes that you want other components
can use, into the 'exports' property of your module definition. Otherwise
your declarations will be private to the module.

**Comment by jusefb** [comment]
Yup, "exports" did the trick. Thank you very much for such a quick and comprehensive response. The NgModules concept starts finally make sense to me

**Comment by spock123** [comment]
Cool dude! Glad to help.

Year initially it's all very confusing.. you will need to play around a bit
before you find the module structure that suits you best.

I don't have the count lf how many times I've refactored my app, lmao

Cheers

**Comment by ocombe** [comment]
Fixed in v2.3.0 ! You can use the new `TranslateModule` (cf docs).
